test/integration/services/modal.js

```diff
@@ -0,0 +1,182 @@
+// import _ from 'underscore';
+
+context('Modal Service', function() {
+  const smallOnTall = () => {
+    cy
+      .getRadio(Radio => {
+        Radio.request('modal', 'show:tall', {
+          headingText: 'Tall Modal Header',
+          bodyText: 'Tall Modal, which is under the small modal.',
+        });
+      });
+
+    cy
+      .getRadio(Radio => {
+        Radio.request('modal', 'show:small', {
+          headingText: 'Small Modal Header',
+          bodyText: 'Small Modal, this guy shows over the top of the tall modal.',
+        });
+      });
+  };
+  specify('Displaying', function() {
+    const buttonStub = cy.stub();
+    cy
+      .visit('/');
+
+    smallOnTall();
+
+    cy
+      .get('.modal--small')
+      .should('contain', 'Small Modal')
+      .find('.modal-cancel')
+      .click()
+      .get('.modal--tall')
+      .should('contain', 'Tall Modal')
+      .find('.modal-cancel')
+      .click();
+
+    // Close using the X icon
+    cy
+      .getRadio(Radio => {
+        Radio.request('modal', 'show:small', {
+          headingText: 'Small Modal Header',
+          bodyText: 'Small Modal',
+          onCancel: buttonStub,
+        });
+      });
+
+    cy
+      .get('.modal--small')
+      .find('.modal-close')
+      .click()
+      .then(() => {
+        expect(buttonStub).to.be.calledOnce;
+        buttonStub.reset();
+      });
+
+    cy
+      .getRadio(Radio => {
+        const modal = Radio.request('modal', 'show:small', {
+          headingText: 'Close On Save',
+          bodyText: 'There\'s a callback to close the modal on save. Also a cool custom class!',
+          buttonClass: 'save-button',
+          className: 'modal--small custom-class',
+          onSubmit: () => {
+            modal.destroy();
+          },
+        });
+      });
+
+    cy
+      .get('.modal--small')
+      .then((modal) => {
+        expect(modal).to.have.class('custom-class');
+      })
+      .find('.save-button')
+      .click()
+      .should('not.exist');
+
+    cy
+      .getRadio(Radio => {
+        Radio.request('modal', 'show', {
+          footerView: false,
+          bodyText: 'No Footer',
+        });
+      });
+
+    cy
+      .get('.modal')
+      .then((modal) => {
+        expect(modal).to.not.have.class('modal-footer');
+      });
+
+    cy
+      .getRadio(Radio => {
+        const modal = Radio.request('modal', 'show', {
+          bodyText: 'Submit Disabled',
+          onSubmit: () => {
+            modal.disableSubmit(true);
+          },
+        });
+      });
+
+    cy
+      .get('.modal')
+      .find('.js-submit')
+      .click()
+      .should('be.disabled');
+
+    cy
+      .getRadio(Radio => {
+        Radio.request('modal', 'show', {
+          onSubmit: buttonStub,
```

Comment:
so while I did say a stub would work here, we do need to add one test to check the default handler.. mainly for code coverage.. and not really because the test is suuuper needed, but more so for watching coverage reports